src/Mindee/Input/Compressor.cs

```diff
@@ -0,0 +1,249 @@
+        {
+            canvas.DrawBitmap(resizedBitmap, SKPoint.Empty);
+
+            if (!keepSourceText)
+            {
+                return;
+            }
+
+            var characters = CheckCharacters(logger, pageReader.GetCharacters(), ref hasWarned);
+
+            foreach (var character in characters)
+            {
+                WriteTextToCanvas(resizedBitmap, character, canvas);
+            }
+        }
+
+
+        /// <summary>
+        /// Generates a bitmap of the current read page. This operation rasterizes the contents.
+        /// </summary>
+        /// <param name="imageQuality">Target quality for the contents of the page.</param>
+        /// <param name="pageReader">Page reader instance for the currently read page.</param>
+        /// <param name="width">Width of the page.</param>
+        /// <param name="height">Height of the page.</param>
+        /// <returns>A resized bitmap of the contents of the rasterized page.</returns>
+        private static SKBitmap GeneratePageBitmap(int imageQuality, IPageReader pageReader, int width, int height)
+        {
+            SKBitmap resizedBitmap = null;
+            try
+            {
+                var rawBytes = pageReader.GetImage();
+                var initialBitmap =
+                    MindeeInputUtils.ArrayToImage(MindeeInputUtils.ConvertTo3DArray(rawBytes, width, height));
+
+                var compressedImage = CompressImage(initialBitmap, imageQuality, width, height);
+
+                var colorType = SKColorType.Argb4444;
+                using var compressedBitmap = SKBitmap.Decode(compressedImage);
+                if (imageQuality > 85)
+                {
+                    colorType = SKColorType.Rgb565;
+                }
+
+                using var surface = SKSurface.Create(new SKImageInfo(width, height, colorType));
+
+                surface.Canvas.DrawBitmap(compressedBitmap, 0, 0);
+                resizedBitmap = SKBitmap.FromImage(surface.Snapshot());
+
+                return resizedBitmap;
+            }
+            catch
+            {
+                resizedBitmap?.Dispose();
+                throw new MindeeInputException("The extracted bitmap from the given object could not be resized.");
+            }
+        }
+
+
+        /// <summary>
+        /// Writes the source text of a page to the newly-created canvas (on top of images).
+        /// Does not extract font-family due to Skiasharp limitations.
+        /// Also approximates the positioning of letters since vertical anchor is not extracted from the text and
+        /// SkiaSharp places letters from bounding boxes.
+        /// </summary>
+        /// <param name="bitmap">The input bitmap.</param>
+        /// <param name="character">The currently read character.</param>
+        /// <param name="canvas">The canvas of the page to insert the bitmap into.</param>
+        private static void WriteTextToCanvas(SKBitmap bitmap, Character character, SKCanvas canvas)
```

Comment:
text should be left as text, not rasterized

Reply:
Very edge case, I'll make it not do so by default.